hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/UpsertPartitioner.java

```diff
@@ -140,11 +140,21 @@ private void assignInserts(WorkloadProfile profile, HoodieEngineContext context)
     Map<String, List<SmallFile>> partitionSmallFilesMap =
         getSmallFilesForPartitions(new ArrayList<String>(partitionPaths), context);
 
+    // get the in pending clustering fileId for each partition path
```

Comment:
Maybe this can also be moved into UpdateStrategy? i.e., UpdateStrategy#filterSmallFiles(List<SmallFile> smallFiles) removes small files  in pending clustering.

Reply:
done